src/dmd/json.d

```diff
@@ -361,22 +368,21 @@ public:
             arrayStart();
             while (stc)
             {
-                const(char)* p = stcToChars(stc);
-                assert(p);
+                auto p = stcToString(stc);
+                assert(p.length);
                 item(p);
             }
             arrayEnd();
         }
     }
 
-    void property(const(char)* linename, const(char)* charname, Loc* loc)
+    extern(D) void property(const(char)[] linename, const(char)[] charname, Loc* loc)
     {
         if (loc)
         {
-            const(char)* filename = loc.filename;
-            if (filename)
+            if (auto filename = toDString(loc.filename))
             {
-                if (!this.filename || strcmp(filename, this.filename))
+                if (!this.filename.length || filename != this.filename)
```

Comment:
I'm guessing you're trying keep the new to code be as close as possible to the old code but this first expression (`!this.filename.length`) isn't needed anymore?